arch/cortex-m/src/support.rs

```diff
@@ -54,3 +55,14 @@ where
 {
     unimplemented!()
 }
+
+pub fn reset() -> ! {
+    unsafe {
+        scb::reset();
+    }
+    loop {
+        // This is required to avoid the empty loop clippy
+        // warning #[warn(clippy::empty_loop)]
+        nop();
+    }
+}
```

Comment:
I think that scb::reset should also have the signature `fn () -> !` and the nop loop should be moved into there.

Reply:
Sorry hudson